.github/workflows/test-pr.yml

```diff
@@ -169,6 +169,52 @@ jobs:
         run: |
           docker stop --time=0 kevm-ci-foundry-${GITHUB_SHA}
 
+  test-prove-foundry-booster:
+    name: 'Build and Test KEVM Foundry proofs (booster)'
+    needs: test-concrete-execution
+    runs-on: [self-hosted, linux, huge]
+    timeout-minutes: 150
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: 'Get booster version'
+        run: |
+          echo "BOOSTER_COMMIT=$(cat deps/booster_release)" >> $GITHUB_ENV  
+      - name: 'Check out booster'
+        uses: actions/checkout@v3
+        with:
+          repository: runtimeverification/hs-backend-booster
+          token: ${{ secrets.JENKINS_GITHUB_PAT }}
+          ref: ${{ env.BOOSTER_COMMIT }}
+          path: booster
+      - name: 'Build booster'
+        run: |
+          cd booster
+          GC_DONT_GC=1 nix build --extra-experimental-features 'nix-command flakes' --print-build-logs .#booster
+          result/bin/booster -h
+      - name: 'Build devShell'
+        env: 
+          CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_PUBLIC_TOKEN }}
+        run: cachix watch-exec k-framework -- nix build --extra-experimental-features 'nix-command flakes' .#devShell.x86_64-linux.inputDerivation --no-link
+      - name: 'Build blockchain-k-plugin-deps'
+        run: nix develop --extra-experimental-features 'nix-command flakes' --command make plugin-deps
+      - name: 'Build kevm-pyk'
+        run: nix develop --extra-experimental-features 'nix-command flakes' --command make poetry
+      - name: 'Build kevm'
+        run: nix develop --extra-experimental-features 'nix-command flakes' --command make build-llvm -j`nproc` RELEASE=true
+      - name: 'Build Foundry'
+        run: nix develop --extra-experimental-features 'nix-command flakes' --command make build-foundry -j2
+      - name: 'Foundry Kompile'
+        run: nix develop --extra-experimental-features 'nix-command flakes' --command make test-foundry-kompile -j2 KEVM_OPTS=--with-llvm-library
+      - name: 'Foundry Prove'
+        run: |
+          BOOSTER_BIN="$(pwd)/booster/result/bin/booster"
+          nix develop --extra-experimental-features 'nix-command flakes' --command make test-foundry-prove -j2 FOUNDRY_PAR=12 KPROVE_OPTS="--use-booster $BOOSTER_BIN"
+      - name: 'Cleanup'
+        run: nix develop --extra-experimental-features 'nix-command flakes' --command make clean
```

Comment:
not sure if this step is necessary? we shouldn't be persisting the `.build` folder afaik, given that it lives in the checked out repo folder.

Reply:
It shouldn't be necessary I don't think.